### PR TITLE
Feat/s1.6.3 request for leave

### PR DESCRIPTION
**Pull request by kirsty-unosquare**
Requesting holidays is now working. The date picker looks horrible but gonna separate it into another branch, a lot of work still to do on it. But in this PR:
 - new modal wrapper within common
 - Booking modal component, shows after selecting a day on calendar.
 - New datepicker currently working only for ios devices. Will be continuing on another branch.
 - Logic for the requesting holidays.

### CODE REVIEW

File: NativeApp/src/components/Home/container.js [hunk 1]
```diff
@@ -18,20 +21,112 @@ export default Container => class extends Component {
...
+        booking: {
+          ...prevState.booking,
+          endDate: formatEndDate,
+        },
+      }));
+    }
+
+    closeModal = () => {
+      const { showModal } = this.state;
+      this.setState({ showModal: !showModal });
```
Comment (d1wilko) [inline NativeApp/src/components/Home/container.js]: should we change `!showModal` to `false` here?
I appreciate that if the modal is open `showModal` should be `true` - but I'm just wary incase future code messes that state setting up.

File: NativeApp/src/components/Home/container.js [hunk 2]
```diff
@@ -18,20 +21,112 @@ export default Container => class extends Component {
...
+      dates.push(endDate);
+
+      return dates;
     }
 
     formatDate = data => data.reduce((obj, item) => {
       const holidayStatus = this.holidayStatus(item.holidayStatusId);
-      obj[item.start] = { textColor: 'white', color: holidayStatus };
+      const dates = this.enumerateDaysBetweenDates(item.start, item.end);
+      for (let i = 0; i < dates.length; i += 1) {
```
Comment (d1wilko) [inline NativeApp/src/components/Home/container.js]: could this not be done as a foreach loop? much nicer syntax

**Review by d1wilko — CHANGES_REQUESTED** [review]
Nice code - just 2 suggestions

File: NativeApp/src/components/Home/container.js (hunk 1 above)
Comment (kirsty-unosquare) [reply NativeApp/src/components/Home/container.js]: Good point! Safer to set it to false.

File: NativeApp/src/components/Home/container.js (hunk 2 above)
Comment (kirsty-unosquare) [reply NativeApp/src/components/Home/container.js]: Can't see why it couldn't.

**Comment by kirsty-unosquare** [comment]
@d1wilko Got those changes up.

**Review by d1wilko — APPROVED** [review]
looks great :)

**Pull request merged**